common/extension/metadata_service_proxy_factory.go

```diff
@@ -0,0 +1,46 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package extension
+
+import (
+	"fmt"
+)
+
+import (
+	"github.com/apache/dubbo-go/metadata/service"
+)
+
+var (
+	metadataServiceProxyFactoryMap = make(map[string]func() service.MetadataServiceProxyFactory)
```

Comment:
hi, @flycash, there are two problems here. firstly, we should init the map as follows,

```Go
metadataServiceProxyFactoryMap = make(map[string]func() service.MetadataServiceProxyFactory， 4)
```

secondly, we should define a func type as follows

```Go
type MetadataServiceProxyFactoryFunc func() service.MetadataServiceProxyFactory
```

And then, u can use this func in this package to simplify codes.